app/javascript/packs/feedPreviewCards.jsx

```diff
@@ -0,0 +1,109 @@
+import { h, render } from 'preact';
+import { UserMetadata } from '../profilePreviewCards/UserMetadata';
+import { initializeDropdown } from '@utilities/dropdownUtils';
+
+async function populateMissingMetadata(metadataPlaceholder) {
+  const { authorId } = metadataPlaceholder.dataset;
+  const response = await fetch(`/profile_preview_cards/${authorId}`, {
+    method: 'GET',
+    headers: {
+      Accept: 'application/json',
+      'X-CSRF-Token': window.csrfToken,
+      'Content-Type': 'application/json',
+    },
+    credentials: 'same-origin',
+  });
+  const authorMetadata = await response.json();
+
+  // A given author may have multiple cards in the feed - populate their metadata everywhere it is missing
+  const allPlaceholdersForThisAuthor = document.querySelectorAll(
+    `.author-preview-metadata-container[data-author-id="${authorId}"]`,
+  );
+
+  for (const placeholder of allPlaceholdersForThisAuthor) {
+    render(
+      <UserMetadata {...authorMetadata} />,
+      placeholder.parentElement,
+      placeholder,
+    );
+
+    placeholder.parentElement.parentElement.style.setProperty(
```

Comment:
Refactor (potentially blocking): If we know the selector for `placeholder.parentElement.parentElement`, prefer `placeholder.closest('some-container-selector')`